core/src/main/scala/org/apache/spark/scheduler/cluster/ClusterInfo.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler.cluster
+
+import java.text.SimpleDateFormat
+
+import scala.collection.mutable.{HashMap}
+
+import org.apache.hadoop.yarn.api.records.{NodeState => YarnNodeState}
+
+/**
+ * State of the node.
+ * Add the node state depending upon the cluster manager, For Yarn
+ * getYarnNodeState is added to create the node state for Decommission Tracker
+ */
+private[spark] object NodeState extends Enumeration {
+  val RUNNING, DECOMMISSIONED, GRACEFUL_DECOMMISSIONING, DECOMMISSIONING, LOST, OTHER = Value
```

Comment:
Is there a reason we aren't adding this to an existing place where tracking state?

Reply:
This is for the node states used in the decommission tracker, whereas in other places it was for executor State

Reply:
I wonder the same thing. We have an executor monitor, this is for node state, but I wonder if they shouldn't be linked better. Also why is this file called ClusterInfo?  Couldn't there also be other states like terminating.  Perhaps we need a node monitor, but not sure if that overlaps to much with the resource-manager pieces.
But I'll have to think about it more

Reply:
document differences between GRACEFUL_DECOMMISSIONING and DECOMMISSIONING

Reply:
1) Above states of the yarn right now is needed here graceful decommission of nodes and the if there some other states that can be considered as other.
2) document differences between GRACEFUL_DECOMMISSIONING and DECOMMISSIONING - DECOMMISSIONING is the yarn node state and GRACEFUL_DECOMMISSIONING we have used to for decommission tracker. Will keep on only one state here to avoid confusion i.e. DECOMMISSIONING